### PR TITLE
[#371] Add functionality for new keyboard expanded layout

### PR DESCRIPTION
**Pull request by damien-rivet**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

This PR implements #371 by adding functionality for the new expanded layout that was introduced by #352.

The `indent` key is now fully functional as well as the `capslock` key.

#### Recordings

**Old design**

https://github.com/scribe-org/Scribe-iOS/assets/1595177/6a6765a9-b749-4554-a032-5fcba6e322dd

**New design**

https://github.com/scribe-org/Scribe-iOS/assets/1595177/4a0e4c1b-f187-4450-9976-7ad333543457

#### Screenshots

| Description | Screenshot |
|--------|--------|
| **New design (iPad)** | Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-10-22 at 14 46 46 |
| **Design on iPhone (no changes)** | Simulator Screenshot - iPhone 15 Pro - 2023-10-22 at 14 54 17 |
| **Capslock** | Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-10-22 at 14 46 53 |
| **Uppercased suggestion** | Simulator Screenshot - iPad Pro (12 9-inch) (6th generation) - 2023-10-22 at 14 56 09 |

### Related issue

- #352

### CODE REVIEW

**Comment by henrikth93** [comment]
Tested it and it looks good to me.

**Comment by andrewtavis** [comment]
Really appreciate the overall care/detail here and especially some a classic sentence for the example, @damien-rivet! 🦊😊 Going through it all now. Sorry that this took a bit. I needed to switch contexts for another project to do UI/UX design, and that always tends to slow down other work (although fun 🙃).

File: Keyboards/KeyboardsBase/KeyAnimation.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyAnimation.swift:322]: Very intuitive :)

File: Keyboards/KeyboardsBase/KeyboardKeys.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardKeys.swift:262]: Appreciate you fixing these formatting errors! 🙏

File: Keyboards/KeyboardsBase/KeyboardKeys.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardKeys.swift:288]: Minor nit that I'm noting here is that we'd like this to be a complete sentence with a period :) I can get it though!

File: Keyboards/KeyboardsBase/KeyboardStyling.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardStyling.swift:43]: Again the dedication here is amazing. Looking into and finding `keysThatAreSlightlyLarger` is great, but also has me worried about the state of some of the code that we're using so many random variables everywhere.

A basic ask would be if you had ideas for how we can improve the code base, it'd be great if you could open some issues. Anything that you've seen that might be some solid refactoring would be very welcome!

File: Keyboards/KeyboardsBase/KeyboardViewController.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardViewController.swift:409]: Interesting, we were missing shift?? Great catch :)

File: Keyboards/KeyboardsBase/KeyboardViewController.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardViewController.swift:1941]: Noting the comment for myself.

File: Keyboards/KeyboardsBase/KeyboardViewController.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardViewController.swift:2502]: Nice as well as we now have both states to account for :)

File: Keyboards/KeyboardsBase/KeyboardViewController.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardViewController.swift:2687]: Another thing that we've been dealing with is that not enough of the code base is split into functions. I do some rounds from time to time and break some things out, but if you came across some parts where we can extract some code then by all means let me know or open an issue!

File: Keyboards/KeyboardsBase/KeyboardKeys.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardKeys.swift:320]: We need the else back here for one bug I'm about to mention such that we get `backgroundColor = specialKeyColor`.

File: Keyboards/KeyboardsBase/KeyboardKeys.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/KeyboardKeys.swift:304]: Losing this line is disabling an indication that we're in caps lock on iPhones :)

**Comment by andrewtavis** [comment]
Two things I'm seeing based on testing:

- For iPhones we now have that the shift key doesn't change to the caps lock key when it's double tapped
  - The caps lock functionality still works, but there's no change in the button to indicate that it will
  - See [this comment](https://github.com/scribe-org/Scribe-iOS/pull/373#discussion_r1375560952)
- The return key is changing to Scribe blue when we're in command mode, but it's white instead of grey for a special key when we're in the normal keyboard functionality (see screens)
  - See [this comment](https://github.com/scribe-org/Scribe-iOS/pull/373#discussion_r1375560671)

Aside from this, really amazing work and one of our best first time contributions, @damien-rivet! I'm happy to get to the final fixes myself. Thank you!

File: Keyboards/KeyboardsBase/KeyboardStyling.swift
Comment (damien-rivet) [reply Keyboards/KeyboardsBase/KeyboardStyling.swift:43]: Good point 🤔 I'll do a quick sweep over the codebase and open a few issues for code quality ^^

File: Keyboards/KeyboardsBase/KeyboardViewController.swift
Comment (damien-rivet) [reply Keyboards/KeyboardsBase/KeyboardViewController.swift:409]: Yup, suggestions were not handled properly for capitalised first-letter but it will be history after the merge.

File: Keyboards/KeyboardsBase/KeyboardKeys.swift
Comment (damien-rivet) [reply Keyboards/KeyboardsBase/KeyboardKeys.swift:320]: There's a default case which should handle the `else`but I'll do a review of my code to see if it's not missing some cases.

**Comment by damien-rivet** [comment]
@andrewtavis I've fixed both issues in my last commit.

While testing I discovered a crash (nil unwrapping) in Command mode, I've added some checks to prevent those from happening again.

File: Keyboards/KeyboardsBase/KeyboardStyling.swift
Comment (andrewtavis) [reply Keyboards/KeyboardsBase/KeyboardStyling.swift:43]: Thanks so much, @damien-rivet! Always looking for some `good first issues` for new folks to work on 😊

**Comment by andrewtavis** [comment]
Fantastic, @damien-rivet! I'm off to dinner, but will get to the final review this evening :) :)

**Review by andrewtavis — APPROVED** [review]
Thanks for getting to the changes so quickly, @damien-rivet, as well as finding and fixing the nil unwrapping! Really great to have this contribution :) Will bring it in and check to make sure that we have everything updated as far as the changelog and such, and then will close out the issue 😊

**Pull request merged**
